diff --git a/effect-schema/column.cjs b/effect-schema/column.cjs
index 08dbca5488666671e2f88b20539fa4ddcd8d13b6..ead0cf0ec7e4970d673380b078713ee21de9813e 100644
--- a/effect-schema/column.cjs
+++ b/effect-schema/column.cjs
@@ -20,3 +20,3 @@ const jsonSchema = effect.Schema.Union([
 ]);
-const bufferSchema = effect.Schema.instanceOf(Buffer);
+const bufferSchema = typeof Buffer === "undefined" ? effect.Schema.Uint8Array : effect.Schema.instanceOf(Buffer);
 function columnToSchema(column) {
diff --git a/effect-schema/column.js b/effect-schema/column.js
index 2574944e1b089187dbf4e3acb42fa99e52674dfd..7b8518fda9cde0a70ce1cb4761dd5fa8bcc65a5c 100644
--- a/effect-schema/column.js
+++ b/effect-schema/column.js
@@ -18,3 +18,3 @@ const jsonSchema = Schema.Union([
 ]);
-const bufferSchema = Schema.instanceOf(Buffer);
+const bufferSchema = typeof Buffer === "undefined" ? Schema.Uint8Array : Schema.instanceOf(Buffer);
 function columnToSchema(column) {
